components/brave_wallet/browser/zcash/zcash_create_orchard_to_orchard_transaction_task.cc

```diff
@@ -35,6 +35,28 @@ ZCashCreateOrchardToOrchardTransactionTask::
 ZCashCreateOrchardToOrchardTransactionTask::
     ~ZCashCreateOrchardToOrchardTransactionTask() = default;
 
+// static
+base::expected<void, mojom::ZCashAddressError>
+ZCashCreateOrchardToOrchardTransactionTask::ValidateAddress(
+    bool testnet,
+    const std::string& addr) {
+  if (!IsUnifiedAddress(addr)) {
+    return base::unexpected(mojom::ZCashAddressError::InvalidUnifiedAddress);
+  }
+
+  if (IsUnifiedTestnetAddress(addr) != testnet) {
+    return base::unexpected(
+        mojom::ZCashAddressError::InvalidAddressNetworkMismatch);
+  }
+
+  if (!ExtractOrchardPart(addr, testnet)) {
+    return base::unexpected(
+        mojom::ZCashAddressError::InvalidUnifiedAddressMissingOrchardPart);
+  }
+
+  return base::ok();
+}
+
```

Comment:
It's the same as for ZCashCreateTransparentToOrchardTransactionTask. Maybe makes sense to move from tasks at all? To zcash_utils for example